### ISSUE TITLE
Pattern for autopublish

### ISSUE THREAD

**Issue by zimme** [MeteorPackaging/underscore.string#1]
We need to find a good pattern for this package so we can try and get autopubblish support going for this package.

What I mean is, a package.js and a meteor.js file maybe with meteor.js file containing the exports stuff needed to setup this package for client and server.

**Comment by zimme** [comment]
With "normal" packages we only need the package.js file and we're good to go but with this we need to export to the right namespace and use npm package server side and so on.

**Comment by splendido** [comment]
:+1:

I'm playing with the folowing `package.js`: 

```javascript
// package metadata file for Meteor.js

Package.describe({
  "name": "underscorestring:underscore.string",
  "summary": "underscore.string (official): String manipulation extensions for Underscore.js javascript library.",
  "version": "2.4.0",
  "git": "https://github.com/epeli/underscore.string.min.git"
});

Package.onUse(function (api) {
  api.versionsFrom("METEOR@1.0");
  api.addFiles("dist/underscore.string.js");
});
```

(there's no `dist/underscore.string.js` file committed to the repo unfortunately...)

...and the result is that `s` gets available as expected bot on the client and on the server

so you can normally do:

```javascript
var myText = s("   underscore string  ").trim().titleize().value();
myText = s.trim("  pippo ");
...
```

wouldn't this be enough for a very lean PR to the official repo?

The use of `s` as a global is what is reported also on the official docs (see [here](http://epeli.github.io/underscore.string/#others)

and the only other lines to change to hook into their build system whould be [these one](https://github.com/MeteorPackaging/underscore.string/blob/meteor-integration/gulpfile.js#L36-40) which should be:

```javascript
gulp.task('bump-in-js', function(semver) {
  return gulp.src([SRC, 'package.js'])
    .pipe(replace(/(version?'?\s?=?\:?\s\')([\d\.]*)\'/gi, '$1' + semver + "'"))
    .pipe(gulp.dest('./'));
});
```

in order to pick up also the version line inside the `pacakge.js` file.
(actually this would fix a bug they have at the moment: the current regexp picks up [this line](https://github.com/MeteorPackaging/underscore.string/blob/meteor-integration/index.js#L6) but not [this one](https://github.com/MeteorPackaging/underscore.string/blob/meteor-integration/index.js#L16).


I have the commit ready if you think this is worth a PR to the official repo ;-)

**Comment by zimme** [comment]
As the official documentation don't recommend using the integration with underscore/lodash I'm fine with this solution.

BUT don't hard code `[SRC, 'package.js']` into `bump-in-js` task. Instead add `package.js` to `VERSION_FILES` at the top of the gulpfile.

The official repo actually has a `dist/underscore.string.js` file available.

**Comment by splendido** [comment]
mmm, `VERSION_FILES` go through `gulp-bump` which expects JSON files...
I've put a new `METEOR = 'package.js;` line at the top of the file and turned the task into:

```javascript
gulp.task('bump-in-js', function(semver) {
  return gulp.src([SRC, METEOR])
    .pipe(replace(/(version?'?\s?=?\:?\s\')([\d\.]*)\'/gi, '$1' + semver + "'"))
    .pipe(gulp.dest('./'));
});
```

is this any better?

**Comment by zimme** [comment]
```js
VERSION_FILES_JS = ['./index.js', 'package.js'];

gulp.task('bump-in-js', function(semver) {
  return gulp.src(VERSION_FILES_JS)
    .pipe(replace(/(version?'?\s?=?\:?\s\')([\d\.]*)\'/gi, '$1' + semver + "'"))
    .pipe(gulp.dest('./'));
});
```

**Comment by splendido** [comment]
:+1:

**Comment by splendido** [comment]
@chrismbeckett are you also fine with this?

See [the comparison](https://github.com/MeteorPackaging/underscore.string/compare/epeli:master...meteor-integration)

**Comment by chrismbeckett** [comment]
Underscore.string is in a transition from being a mixin for underscore, to being a stand-alone library. As  such, the documentation regarding what the export should be was a bit confusing when I originally forked this. In one location I found them refer to the "s" global, but in samples and function details they demonstrated "_s" as the global. I went with the "_s" to preserve its roots related to underscore. I can't seem to find the documentation where I found them using "_s", but it does seem like the docs all now show using the "s" global.

**Comment by chrismbeckett** [comment]
No, I have no problems with any changes you want to make to support autopublish. I have to admit that I have ongoing concerns about trying to get original authors to support Meteor directly. I have not submitted pull requests for the wrapped packages I have originally published waiting for either your autopublish to mature, or the idea of doing pull requests to the original authors is abandoned.

**Comment by splendido** [comment]
Ok, good :)
I could do the PR, but if you're willing to feel free to do it yourself @chrismbeckett (See the [wiki](https://github.com/MeteorPackaging/autopublish.meteor.com/wiki))!

**Comment by splendido** [comment]
...ok, I've just done it ;-)
